### DISCUSSION TITLE
Is there an easy way to clear area buffer?

### DISCUSSION THREAD

**Discussion by Berkays** [olikraus/u8g2#1719] [Q&A] [answered]
I'm using UpdateDisplayArea to display individual components on the oled screen. Is there an easier way than painting the region with inverse color then going with UpdateDisplayArea? I was thinking a procedure like ClearDisplayArea could be useful.

**Comment by olikraus** [comment, marked as answer]
UpdateDisplayArea acts more like a post processing filter: Only pixel data within the specified rectangle will be uploaded to the display.
Bitmap data on the display outside the specified rectangle is just kept as it is. 
I mean, you should have some code like this:

```
  u8g2.setFont(u8g2_font_ncenB14_tr);
  u8g2.firstPage();
  do {
    u8g2.setCursor(0, 20);
    u8g2.print(F("Hello World!"));
  } while ( u8g2.nextPage() );
```

If you just want to update a part of the display (for performance reasons for example), then you just keep the loop, but prefix this with UpdateDisplayArea:

```
  u8g2.UpdateDisplayArea(...);
  u8g2.setFont(u8g2_font_ncenB14_tr);
  u8g2.firstPage();
  do {
    u8g2.setCursor(0, 20);
    u8g2.print(F("Hello World!"));
  } while ( u8g2.nextPage() );
```
Clearing the picture will automatically happen with the firstPage/nextPage.

Let me say it like this: "UpdateDisplayArea" is not a pixel drawing function, it is more like a configuration option to restrict the access of the drawing functions.

**Reply by Berkays** [reply]
I understand. I made a wrapper for my case. Thanks for the answer.
